src/main/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidator.java

```diff
@@ -39,7 +40,31 @@ public SubmissionValidator(SubmissionOptions options) {
     }
 
     public void validate() throws ValidationEngineException {
-        new SubmissionValidationPlan(options).execute();
+
+        ValidationResult validationResult = new SubmissionValidationPlan(options).execute();
+        if(!validationResult.isValid()) {
+
+            if(options.isRemote) {
+                writeErrorText(validationResult.getDefaultOrigin().getOriginText());
+                throw new ValidationEngineException(validationResult.getDefaultOrigin().getOriginText(), ValidationEngineException.ReportErrorType.VALIDATION_ERROR);
+            } else {
+                StringBuilder sb = new StringBuilder();
+                for (ValidationMessage<Origin> error : validationResult.getMessages(Severity.ERROR)) {
+                    if((sb.length() + error.getMessage().length()) > ERROR_MAX_LENGTH)
+                        break;
+                    sb.append(error.getMessage());
+                    sb.append("\n");
+                }
+                throw new ValidationEngineException(StringUtils.chomp(sb.toString()), ValidationEngineException.ReportErrorType.VALIDATION_ERROR);
+            }
```

Comment:
When we are running locally do we really need to concatenate the full error message to the ValidationEngineException?

Reply:
This is only for assembly pipeline, this block of appending some error message won't be executed for Webin-CLI. Also, the length is restricted so we won't iterate all the exceptions.

Reply:
Ok, thanks. We have some kind of error summary thing going on here.